### ISSUE TITLE
NPE during modified unit test

### ISSUE THREAD

**Issue by spyhunter99** [mucaho/jnetrobust#14]
```
[0] 40, 60000, 0.75, 0.25, 16, 1000, true (testDelayed)(com.github.mucaho.jnetrobust.DelayedTest)  Time elapsed: 1,472.111 sec  <<< ERROR!
java.lang.NullPointerException
    at com.github.mucaho.jnetrobust.controller.Controller.receive(Controller.java:92)
    at com.github.mucaho.jnetrobust.controller.Controller.receive(Controller.java:86)
    at com.github.mucaho.jnetrobust.util.TestHost.consume(TestHost.java:73)
    at com.github.mucaho.jnetrobust.util.TestHost.receive(TestHost.java:64)
    at com.github.mucaho.jnetrobust.util.TestHost.receive(TestHost.java:59)
    at com.github.mucaho.jnetrobust.DelayedTest.testDelayed(DelayedTest.java:202)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at java.lang.reflect.Method.invoke(Method.java:606)
    at junitparams.internal.InvokeParameterisedMethod.evaluate(InvokeParameterisedMethod.java:209)
    at junitparams.internal.ParameterisedTestMethodRunner.runMethodInvoker(ParameterisedTestMethodRunner.java:49)
    at junitparams.internal.ParameterisedTestMethodRunner.runTestMethod(ParameterisedTestMethodRunner.java:42)
    at junitparams.internal.ParameterisedTestClassRunner.runParameterisedTest(ParameterisedTestClassRunner.java:143)
    at junitparams.JUnitParamsRunner.runChild(JUnitParamsRunner.java:393)
    at junitparams.JUnitParamsRunner.runChild(JUnitParamsRunner.java:371)
    at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
    at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
    at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at java.lang.reflect.Method.invoke(Method.java:606)
    at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
    at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
    at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
    at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
    at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
```

**Comment by mucaho** [comment]
Thanks for testing under these extreme conditions. This has identified a bug:
Data that hasn't received an ACK is queued for retransmission, but is subsequently removed due to the restrictions (e.g. size and/or time limits) of the internal queue.
However, due to the incremental nature of the implementation, the data is only removed from the queue, but not from the resulting packet that will be sent, leading to inconsistent state.

**Comment by mucaho** [comment]
Apologies for the very late response, but this should have been resolved by ea1c035e95356cfb74a1f918426f652b50a85e1c. Please reopen if you still use the library and experience the issue!
